src/aws/compute/method.ts

```diff
@@ -624,7 +635,10 @@ export class Method extends AwsConstructBase {
           integrationId,
           this.renderIntegration(this.bindResult),
         );
-      deployment?.node.addDependency(integrationResource);
+      // call node.addDependency on every deployment tracked
+      for (const deployment of this._deployments) {
+        deployment.node.addDependency(integrationResource);
+      }
```

Comment:
interesting fix, previously only the `latestDeployment` from the `api` would get the dependency, but now all deployments will only be created after the method have their integration resources successfully created. Fixing more of the race conditions observed.